{{cookiecutter.python_name}}/jupyter_server_test_config.py

```diff
@@ -0,0 +1,17 @@
+"""Server configuration for integration tests.
+
+!! Never use this configuration in production because it
+opens the server to the world and provide access to JupyterLab
+JavaScript objects through the global window variable.
+"""
+from tempfile import mkdtemp
+
+c.ServerApp.port = 8888
+c.ServerApp.port_retries = 0
+c.ServerApp.open_browser = False
+
+c.ServerApp.root_dir = mkdtemp(prefix='galata-test-')
```

Comment:
this won't clean up after itself

Reply:
The setup is done like that to mimic pytest system (without the auto clean-up feat though).

The UI tests can easily fail if the filebrowser is crowded by existing files (due to past test runs or previous tests). To mitigate that problem, galata is creating a folder for each test (using the test name). That folder and its content will be deleted in the teardown step. So the temporary folder is likely to end up empty at the end of a test run; except if an error occurs. If an error occurs, the files are not removed (default behavior). So they can be access afterwards for introspection. That behavior is controlled by a [fixture](https://github.com/jupyterlab/jupyterlab/tree/master/galata#serverfiles).

So I don't see any problem crowding the temporary folder as per definition it is the first place to clean if you end up having low disk space. And this is much easier than going through tens of projects folder for removing test artifacts.